### PR TITLE
Add minimal App Engine support

### PR DESCRIPTION
**Pull request by jamiesnape**
Currently only for synchronous submissions and with RSS disabled.

### CODE REVIEW

**Comment by jamiesnape** [comment]
PTAL @cpatrick, @danlamanna, @zackgalbreath.

File: include/ctestparser.php [hunk 1]
```diff
@@ -93,15 +93,18 @@ function writeBackupFile($filehandler, $content, $projectname, $buildname,
...
+            if (realpath($CDASH_DATA_ROOT_DIRECTORY) !== false) {
+                // Make sure the file is in the right directory.
+                $pos = strpos(realpath(dirname($filename)), realpath($backupDir));
+                if ($pos === false || $pos != 0) {
+                    echo "File cannot be stored in backup directory: $filename";
+                    add_log("File cannot be stored in backup directory: $filename (realpath = " . realpath($backupDir) . ')', 'writeBackupFile', LOG_ERR);
+                    flock($lockfp, LOCK_UN);
+                    unlink($lockfilename);
+                    return false;
+                }
```
Comment (zackgalbreath) [inline include/ctestparser.php]: This means that we currently don't backup the submitted XML files when running on GAE, right?

Comment (jamiesnape) [inline include/ctestparser.php]: Correct, for now.

**Comment by zackgalbreath** [comment]
In general this LGTM.  I love the removal of all the old PHP code.

As we work to improve query performance for GAE we should also revisit whether or not all those calls of `set_time_limit(0)` are really necessary.

**Comment by jamiesnape** [comment]
@zackgalbreath PTAL.

**Comment by zackgalbreath** [comment]
New changes (tar extraction & such) LGTM

**Pull request merged**